staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -147,6 +152,17 @@ func (sanitizedAuthConfigPersister) String() string {
 	return "rest.AuthProviderConfigPersister(--- REDACTED ---)"
 }
 
+// RESTClient returns a RESTClient that satisfies the requested attributes on a client Config
+// object. Note that a RESTClient may require fields that are optional when initializing a Client.
+// A RESTClient created by this method is generic - it expects to operate on an API that follows
+// the Kubernetes conventions, but may not be the Kubernetes API.
+func (c *Config) RESTClient() (Interface, error) {
```

Comment:
This increases the responsibility of the configuration substantially; it is now client builder. It is more tightly coupled with `rest.Interface`. I'm not sure it's worth doing just for the minor convenience gained.